ci_scripts/generate_ios26_testplan.rb

```diff
@@ -0,0 +1,245 @@
+        if class_name
+          test_identifier = "#{class_name}/#{method_name}()"
+          log_info "    ↳ Method annotation: #{test_identifier}"
+          @tests_by_target[target_name] << test_identifier
+        else
+          log_warning "    ↳ Could not find class name for method #{method_name}"
+        end
+        break
+      
+      # If we hit something else that's not a comment or empty line, stop looking
+      else
+        log_info "    ↳ @iOS26 not followed by class or test method: #{line}"
+        break
+      end
+    end
+  end
+
+  def find_class_name_for_method(lines, method_index)
+    # Search backwards from method to find the class declaration
+    (0...method_index).reverse_each do |i|
+      line = lines[i].strip
+      if class_match = line.match(/class\s+(\w*Tests?)\s*:\s*\w*TestCase/)
+        return class_match[1]
+      end
+    end
+    nil
+  end
+
+  def add_all_test_methods_from_content(content, class_name, target_name)
+    test_methods = content.scan(/func\s+(test\w*)\s*\(\)/).flatten
+    test_methods.each do |method|
+      @tests_by_target[target_name] << "#{class_name}/#{method}()"
+    end
+    log_info "    ↳ Added #{test_methods.length} test methods from class #{class_name}"
+  end
+
+  def determine_test_target(file_path)
+    # Map file paths to test target names based on directory structure
+    case file_path
+    when /StripePaymentSheet.*Tests/
+      "StripePaymentSheetTests"
+    when /StripeCore.*Tests/
+      "StripeCoreTests"
+    when /StripePayments.*Tests/
+      "StripePaymentsTests"
+    when /StripePaymentsUI.*Tests/
+      "StripePaymentsUITests"  
+    when /StripeUICore.*Tests/
+      "StripeUICoreTests"
+    when /StripeConnect.*Tests/
+      "StripeConnectTests"
+    when /StripeIdentity.*Tests/
+      "StripeIdentityTests"
+    when /StripeFinancialConnections.*Tests/
+      "StripeFinancialConnectionsTests"
+    when /StripeCardScan.*Tests/
+      "StripeCardScanTests"
+    when /StripeApplePay.*Tests/
+      "StripeApplePayTests"
+    when /StripeCameraCore.*Tests/
+      "StripeCameraCoreTests"
+    when /StripeCryptoOnramp.*Tests/
+      "StripeCryptoOnrampTests"
+    when /Stripe3DS2.*Tests/
+      if file_path.include?("DemoUI")
+        "Stripe3DS2DemoUITests"
+      else
+        "Stripe3DS2Tests"
+      end
+    when /Stripe\/.*Tests/, /StripeiOS.*Tests/
+      "StripeiOSTests"
+    else
+      nil
+    end
+  end
+
+  def write_test_plan
+    if File.exist?(@test_plan_file)
+      # Read existing test plan
+      existing_plan = JSON.parse(File.read(@test_plan_file))
+      
+      # Update EVERY test target with selectedTests
+      existing_plan["testTargets"].each do |target|
+        target_name = target["target"]["name"]
+        
+        if @tests_by_target[target_name] && @tests_by_target[target_name].any?
+          # Sort and dedupe tests for this target
+          target["selectedTests"] = @tests_by_target[target_name].sort.uniq
```

Comment:
I'm a little surprised that we would have the potential for duplicated tests -- is this needed?

Reply:
I didn't notice this but I suppose it's reasonable to keep, like it's valid if you do:
```
// @iOS26
class FooTest { 
  // @iOS26
  func testFoo() {
```